src/data/actions/subscriptionDetails.js

```diff
@@ -0,0 +1,58 @@
+import {
+  FETCH_SUBSCRIPTION_DETAILS_REQUEST,
+  FETCH_SUBSCRIPTION_DETAILS_SUCCESS,
+  FETCH_SUBSCRIPTION_DETAILS_FAILURE,
+  CLEAR_SUBSCRIPTION_DETAILS,
+} from '../constants/subscriptionDetails';
+import { fetchSubscriptionDetails as fetchSubscriptionDetailsEndpoint } from '../../components/subscriptions/data/service';
+import NewRelicService from '../services/NewRelicService';
+
+const emptySubscriptionDetails = {
+  uuid: null,
+  purchaseDate: null,
+  startDate: null,
+  endDate: null,
+  licenses: {
+    allocated: 0,
+    available: 0,
+  },
+};
+
+const fetchSubscriptionDetailsRequest = () => ({ type: FETCH_SUBSCRIPTION_DETAILS_REQUEST });
+const fetchSubscriptionDetailsSuccess = data => ({
+  type: FETCH_SUBSCRIPTION_DETAILS_SUCCESS,
+  payload: { data },
+});
+const fetchSubscriptionDetailsFailure = error => ({
+  type: FETCH_SUBSCRIPTION_DETAILS_FAILURE,
+  payload: { error },
+});
+
+const fetchSubscriptionDetails = enterpriseId => (
+  (dispatch) => {
+    dispatch(fetchSubscriptionDetailsRequest());
+    return fetchSubscriptionDetailsEndpoint(enterpriseId)
+      .then((response) => {
+        dispatch(fetchSubscriptionDetailsSuccess(response));
+      })
+      .catch((error) => {
+        NewRelicService.logAPIErrorResponse(error);
+        // This endpoint returns a 404 if no data exists,
+        // so we convert it to an empty response here.
+        if (error.response.status === 404) {
```

Comment:
This logic is fine for now, but I'm not entirely sure at this time how we'll be handling when the specified enterprise doesn't have an active subscription plan (i.e., if we'll be returning a 404 or an empty list). So, let's keep it as is for now and can always come back to modify if necessary 👍